SPEC.md

```diff
@@ -0,0 +1,111 @@
+# adminbot spec
+
+This file documents both what adminbot does ambiently and what commands it responds to.
+
+## Behaviors
+
+### Welcome message
+
+adminbot knows what users it has seen and which ones it hasn't. When an unrecognized user joins
+`#tildetown`, adminbot welcomes them and gives some helpful hints.
+
+```
+--> | wren has joined
+<adminbot> Welcome, wren! I hope you're well. If you haven't seen it yet, please read over our wiki
+           page about this chat: https://tilde.town/wiki/socializing/irc
+```
+
+### Admin PSAs
+
+Admins have use of the `!psa` command which has adminbot anonymously relay a command to `#tildetown`.
+
+```
+<adminbot> ***** PSA: The server is being restarted in T-2 hours. Out your crushes. Hug your plushies.
+```
+
+## Commands
+
+### !rollcall
+
+Reports its purpose and presence.
+
+```
+<wren> !rollcall
+<adminbot> I act as a helper for the town's admins! For more info see https://github.com/tildetown/adminbot .
+```
+
+### !help
+
+Returns a listing of this bot's commands. Should probably only be used in a `/query`.
+
+```
+<wren> i forget what adminbot does
+<wren> but i'll ask it privately so as not to spam
+<wren> /query adminbot !help
+<adminbot> !command0 helptext
+<adminbot> !command1 helptext
+<adminbot> !command2 helptext
+```
+
+### !adminteam
+
+This command lists out the current town admin roster. Please use **sparingly** as it will highlight
```

Comment:
Given that we have `!admins` and `!page` for actually getting admins' attention, perhaps this command should reply in PM as well?
Alternatively we could just use op status in #tildetown for this, and not require this command after all.

Reply:
i'm pro responding with PMs. I don't want to rely on just @ status since ops aren't always lurking in irc but i'd still like them noted as admins.